app/controllers/api/v0/user_roles_controller.rb

```diff
@@ -269,74 +269,77 @@ def index_for_group_type
+    ]
+    group = UserGroup.find(group_id)
+    status = params.require(:status) if UserGroup.group_types_containing_status_metadata.include?(group.group_type)
+    location = params[:location] if group.group_type == UserGroup.group_types[:delegate_regions]
+
+    return render status: :unprocessable_entity, json: { error: "Invalid group type" } unless create_supported_groups.include?(group.group_type)
+    return head :unauthorized unless current_user.has_permission?(:can_edit_groups, group_id)
+
+    if status.present? && group.unique_status?(status)
+      end_role_for_user_in_group_with_status(group, status)
+    end
+
+    if group.group_type == UserGroup.group_types[:delegate_regions] && !delegate_status_migrated?(status)
       # Creates deprecated role.
-      group = UserGroup.find(group_id)
-      return head :unauthorized unless current_user.has_permission?(:can_edit_groups, group_id)
-      status = params.require(:status) if UserGroup.group_types_containing_status_metadata.include?(group.group_type)
-      location = params[:location] if group.group_type == UserGroup.group_types[:delegate_regions]
-      if status.present? && group.unique_status?(status)
-        end_role_for_user_in_group_with_status(group, status)
-      end
       user = User.find(user_id)
       user.update!(delegate_status: status, region_id: group.id, location: location)
       send_role_change_notification(user)
-      render json: {
-        success: true,
-      }
-    else
-      group = UserGroup.find(group_id)
-      return head :unauthorized unless current_user.has_permission?(:can_edit_groups, group_id)
-      create_supported_groups = [
-        UserGroup.group_types[:delegate_regions],
-        UserGroup.group_types[:delegate_probation],
-        UserGroup.group_types[:translators],
-      ]
-      unless create_supported_groups.include?(group.group_type)
-        render status: :unprocessable_entity, json: { error: "Invalid group type" }
-        return
-      end
-      status = params.require(:status) if UserGroup.group_types_containing_status_metadata.include?(group.group_type)
-      if status.present? && group.unique_status?(status)
-        end_role_for_user_in_group_with_status(group, status)
-      end
-      if group.group_type == UserGroup.group_types[:delegate_regions]
-        location = params[:location]
-        metadata = RolesMetadataDelegateRegions.create!(status: status, location: location)
-      end
-      role = UserRole.create!(
-        user_id: user_id,
-        group_id: group_id,
-        start_date: Date.today,
-        metadata: metadata,
-      )
-      RoleChangeMailer.notify_role_start(role, current_user).deliver_later
-      render json: {
+      return render json: {
         success: true,
       }
     end
+
+    if group.group_type == UserGroup.group_types[:delegate_regions]
+      metadata = RolesMetadataDelegateRegions.create!(status: status, location: location)
+    end
```

Comment:
How does the variable `metadata` get declared for other group types?

Reply:
This is not about the high-level logic which group needs metadata. This is about the low-level programming question where the variable is being defined.

In other words, if other groups don't need metadata it should be set to null somewhere.